src-java/kilda-persistence-hibernate/src/main/java/org/openkilda/persistence/hibernate/repositories/HibernateHistoryFlowEventRepository.java

```diff
@@ -0,0 +1,179 @@
+/* Copyright 2021 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.persistence.hibernate.repositories;
+
+import org.openkilda.model.history.FlowEvent;
+import org.openkilda.model.history.FlowEvent.FlowEventCloner;
+import org.openkilda.model.history.FlowEvent.FlowEventData;
+import org.openkilda.model.history.FlowEventAction;
+import org.openkilda.model.history.FlowEventAction.FlowEventActionCloner;
+import org.openkilda.model.history.FlowEventDump;
+import org.openkilda.model.history.FlowEventDump.FlowEventDumpCloner;
+import org.openkilda.model.history.FlowStatusView;
+import org.openkilda.persistence.exceptions.PersistenceException;
+import org.openkilda.persistence.hibernate.entities.history.HibernateFlowEvent;
+import org.openkilda.persistence.hibernate.entities.history.HibernateFlowEventAction;
+import org.openkilda.persistence.hibernate.entities.history.HibernateFlowEventDump;
+import org.openkilda.persistence.hibernate.entities.history.HibernateFlowEvent_;
+import org.openkilda.persistence.repositories.history.FlowEventRepository;
+import org.openkilda.persistence.tx.TransactionManager;
+
+import org.hibernate.SessionFactory;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+
+public class HibernateHistoryFlowEventRepository
+        extends HibernateGenericRepository<org.openkilda.model.history.FlowEvent, FlowEventData, HibernateFlowEvent>
+        implements FlowEventRepository {
+    public HibernateHistoryFlowEventRepository(
+            TransactionManager transactionManager, Supplier<SessionFactory> factorySupplier) {
+        super(transactionManager, factorySupplier);
+    }
+
+    @Override
+    public boolean existsByTaskId(String taskId) {
+        return transactionManager.doInTransaction(() -> findEntityByTaskId(taskId).isPresent());
+    }
+
+    @Override
+    public Optional<org.openkilda.model.history.FlowEvent> findByTaskId(String taskId) {
+        return transactionManager.doInTransaction(
+                () -> findEntityByTaskId(taskId).map(org.openkilda.model.history.FlowEvent::new));
```

Comment:
No need to use fqn here.

Reply:
yes. removing. It was required before #4376